integration-tests/src/test/java/oracle/kubernetes/operator/ItOperatorUpgrade.java

```diff
@@ -238,41 +141,74 @@ public void testOperatorUpgradeFrom2_3_0() throws Exception {
-    if (restart) {
-      checkDomainRollingRestarted();
-    }
-    checkOperatorVersion();
+    checkDomainNotRestarted();
+    checkCrdVersion();
+    // checkDomainApiVersion(); Not Needed 
     testClusterScaling(operator, domain, false);
   }
 
   /**
-   * checks the expected version of the upgraded operator in a loop. In Jenkins it takes nearly 8
-   * minutes to show the updated value of the domain CRD.
+   * Checks Expected/Upgraded ApiVersion of the Domain Object in a loop. 
+   * In Jenkins it takes nearly 8 minutes to show the updated value 
+   * 
+   * The .apiVersion of the domain will not be updated until the domain is 
+   * written again (e.g. by the operator when updating status)
+   * This contributes to long time period. We can skip this check 
    *
    * @throws Exception when version does not match
    */
-  private void checkOperatorVersion() throws Exception {
+  private void checkDomainApiVersion() throws Exception {
     boolean result = false;
     LoggerHelper.getLocal().log(
         Level.INFO,
         "Checking for the domain apiVersion "
             + getDomainApiVersion()
             + " in a loop for up to 15 minutes");
-    for (int i = 0; i < 900; i = i + 10) {
+    for (int i = 0; i < 120; i++) {
       ExecResult exec =
           TestUtils.exec(
               "kubectl get domain -n " + DOM_NS + "  " + DUID + " -o jsonpath={.apiVersion}", true);
       if (exec.stdout().contains(getDomainApiVersion())) {
-        LoggerHelper.getLocal().log(Level.INFO, "Got the expected apiVersion");
+        int ttime = 10 * (i - 1); 
+        LoggerHelper.getLocal().log(Level.INFO, "Got the expected API Version after [" + ttime + "] Seconds");
+        result = true;
+        break;
+      }
+      Thread.sleep(1000 * 10);
+    }
+    if (!result) {
+      throw new RuntimeException("FAILURE: Didn't get the expected API Version");
+    }
+  }
+
+  /**
+   * Checks the expected Upgraded Version of CustomResourceDefintion (CRD) 
+   * in default NameSpace in a loop. 
+   *
+   * @throws Exception when version does not match
+   */
+  private void checkCrdVersion() throws Exception {
+    boolean result = false;
+    LoggerHelper.getLocal().log(
+        Level.INFO,
+        "Checking for the CRD Version "
+            + getDomainApiVersion()
+            + " in a loop for up to 15 minutes");
+    for (int i = 0; i < 900; i = i + 10) {
+      ExecResult exec =
+          TestUtils.exec(
+              "kubectl get crd domains.weblogic.oracle -o jsonpath='{.spec.versions[?(@.storage==true)].name}'", true);
+      if (exec.stdout().contains(getCrdVersion())) {
+        LoggerHelper.getLocal().log(Level.INFO, "Got expected CRD Version");
         result = true;
         break;
       }
       Thread.sleep(1000 * 10);
     }
     if (!result) {
-      throw new RuntimeException("FAILURE: Didn't get the expected operator version");
+      throw new RuntimeException("FAILURE: Didn't get expected CRD Version");
```

Comment:
does this need to be a RuntimeException? not just a regular Exception?

Reply:
Modified as follows  ..
  throw new Exception("FAILURE: Didn't get expected CRD Version");

Reply:
well that's great that you changed it, but the question was does it **need** to be a RE?

Reply:
Junit fails when the method throws any Exception, it does not matter it is a RE or simple Exception.